### PR TITLE
etcd: add backup config controller

### PR DESCRIPTION
**Pull request by hexfusion**
Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

### CODE REVIEW

File: enhancements/etcd/backup-config-controller.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+Backups are an important part of cluster management. Our customers
+expect a safe and easy backup procedure.
+
+## Motivation
+The current cluster backup procedure requires envolking scripts on cluster to
```
Comment (kikisdeliveryservice) [inline enhancements/etcd/backup-config-controller.md:35]: envolking -> involking

File: enhancements/etcd/backup-config-controller.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+perform backup operations. Because we rely upon on disk resources for the backup
+process. Putting the logic
```
Comment (kikisdeliveryservice) [inline enhancements/etcd/backup-config-controller.md:37]: Incomplete sentence?

File: enhancements/etcd/backup-config-controller.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+1.  Eliminate on cluster scripts as a requirement
+1.  Utilize the cluster-etcd-operator to manage the backup process.
+
+## Non-Goals
+
+*  Automated backup is not considered for this enhancement.
+
+## Currently Supported Functionality 
+
+Currently the scripts available support the following functionality:
```
Comment (kikisdeliveryservice) [inline enhancements/etcd/backup-config-controller.md:50]: Should we link to scripts?

File: enhancements/etcd/backup-config-controller.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+The following changes are proposed.
+
+- library-go changes: [PR #855](https://github.com/openshift/library-go/pull/855)
+- etcd-operator changes: [PR #894](https://github.com/openshift/cluster-etcd-operator/pull/406)
+- API changes: TBD
+
+### BackupConfigController
+
+Phase 1 of automated backups is the `BackupConfigController` this controller is tasked with
+populating the spec for a cluster-backup pod. The cluster-backup pod persists the TLS
```
Comment (kikisdeliveryservice) [inline enhancements/etcd/backup-config-controller.md:69]: maybe call out namespace for pod here?

File: enhancements/etcd/backup-config-controller.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+1. Create cluster backup controller that will facilitate populating the nessisary assets.
+1. Review adding customer facing API.
+
+## Risks and Mitigations
+
+* Performance impact
+
+  We would want to limit the frequency of backups to one hour.
+
+* Disks running out of space
```
Comment (kikisdeliveryservice) [inline enhancements/etcd/backup-config-controller.md:196]: Is there a mitigation for this one?

**Comment by jomeier** [comment]
@hexfusion @kikisdeliveryservice 
An easy workflow for Export / Import of etcd backups would be a great enhancement for OpenShift.

**Pull request closed without merging**